src/main/scala/net/snowflake/spark/snowflake/pushdowns/querygeneration/DateStatement.scala

```diff
@@ -22,8 +29,29 @@ private[querygeneration] object DateStatement {
 
     Option(expr match {
       case DateAdd(startDate, days) =>
-        ConstantString("DATEADD(day,") + convertStatement(days, fields) + "," +
-          convertStatement(startDate, fields) + ")"
+        ConstantString(SNOWFLAKE_DATEADD) +
+          blockStatement(
+            ConstantString("day,") +
+              convertStatement(days, fields) + "," +
+              convertStatement(startDate, fields)
+          )
+
+      // Snowflake has no direct DateSub function,
+      // it is pushdown by DATEADD with negative days
+      case DateSub(startDate, days) =>
+        ConstantString(SNOWFLAKE_DATEADD) +
+          blockStatement(
+            ConstantString("day, (0 - (") +
+              convertStatement(days, fields) + ") )," +
+              convertStatement(startDate, fields)
```

Comment:
Your statement is`DATAADD(day, (0 - (number)), date)`
Can we simplify it to `DATAADD(day, - number, date)`

Reply:
No. It doesn't work. For example, number = -1, there is syntax error.
`DATAADD(day, - (number), date)` should work. But, I think `(0 - (number))` is safest.

Reply:
Can we use a negation function on `days`?

Reply:
yes. The test covers the combination on DataAdd/DataSub with +1/-1. For example,
      date_add(col("leap_02_29"),1),  // Add +1
      date_sub(col("leap_02_29"),-1), // Sub -1
      date_add(col("leap_03_01"),-1), // Add -1
      date_sub(col("leap_03_01"),1),  // Sub +1